src/plugins/gandi-solution/index.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from "react";
+import styles from "./styles.less";
+import GandiSolutionIcon from "assets/icon--gandi-solution.svg";
+import PluginsManager from "../plugins-manager/index";
+import ExpansionBox, { ExpansionRect } from "components/ExpansionBox";
+import useStorageInfo from "hooks/useStorageInfo";
+import ReactDOM from "react-dom";
+import Tooltip from "components/Tooltip";
+import Tab from "components/Tab";
+import Home from "./componet/Home";
+import IF from "components/IF";
+import hack from "./hack.js";
+import BluePrintList from "./componet/BluePrintList";
+import ArticleList from "./componet/ArticleList";
+import DemoList from "./componet/DemoList";
+import toast from "react-hot-toast";
+
+const DEFAULT_CONTAINER_INFO = {
+  width: 300,
+  height: 450,
+  translateX: 72,
+  translateY: 60,
+};
+
+const GandiSolution: React.FC<PluginContext> = ({ msg, registerSettings, vm, workspace, blockly, utils }) => {
+  const [visible, setVisible] = React.useState(false);
+  const [currentIndex, setCurrentIndex] = React.useState(0);
+  const [loading, setLoading] = React.useState("load");
+  const [Jump, setJump] = React.useState("");
+  // const [Link, setLink] = React.useState("https://learn.ccw.site");
+  // const [Link, setLink] = React.useState("http://localhost:3000");
+  const [Link, setLink] = React.useState("http://learn-qa.xiguacity.cn");
+
+  const handleClick = React.useCallback((e: React.MouseEvent | undefined) => {
+    if (e) e.preventDefault();
+    setLoading("load");
+    if (e) {
+      const rect = (e.target as HTMLDivElement).getBoundingClientRect();
+      setContainerInfo({
+        ...containerInfoRef.current,
+        translateX: rect.x + 28,
+        translateY: rect.y - 6,
+      });
+    } else {
+      setContainerInfo({
+        ...containerInfoRef.current,
+      });
+    }
+    setVisible(true);
+  }, []);
+
+  React.useMemo(() => {
+    if (window.location.search.includes("Block")) {
+      setLink(`${Link}/article/${window.location.search.split("Block=")[1].split("&")[0]}`);
```

Comment:
A better way to extract the url query is to use https://developer.mozilla.org/en-US/docs/Web/API/URL/URL